app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/RtlUtils.java

```diff
@@ -0,0 +1,466 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import android.util.Log;
+import android.util.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import nodomain.freeyourgadget.gadgetbridge.GBApplication;
+
+class RtlUtils {
+    /**
+     * Checks the status of right-to-left option
+     * @return true if right-to-left option is On, and false, if Off or not exist
+     */
+    public static boolean contextualSupport()
+    {
+        return GBApplication.getPrefs().getBoolean("contextualArabic", false);
+    }
+
+    //map with brackets chars to change there direction
+    private static Map<Character, Character> directionSignsMap = new HashMap<Character, Character>(){
+        {
+            put('(', ')'); put(')', '('); put('[', ']'); put(']', '['); put('{','}'); put('}','{');
+
+
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> hebrewRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            add(new Pair<Character, Character>('\u0590', '\u05F4'));
+            add(new Pair<Character, Character>('\uFB1D', '\uFB4F'));
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> arabicRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            add(new Pair<Character, Character>('\u0600', '\u06FF'));
+            add(new Pair<Character, Character>('\u0750', '\u077F'));
+            add(new Pair<Character, Character>('\u08A0', '\u08FF'));
+            add(new Pair<Character, Character>('\uFB50', '\uFDFF'));
+            add(new Pair<Character, Character>('\uFE70', '\uFEFF'));
+        }
+    };
+
+    //list of unicode ranges of rtl chars
+    private static ArrayList<Pair<Character, Character>> rtlRange = new ArrayList<Pair<Character, Character>>() {
+        {
+            addAll(hebrewRange);
+            addAll(arabicRange);
+        }
+    };
+
+    /**
+     * @return true if the char is in the rtl range, otherwise false
+     */
+    static Boolean isHebrew(char c){
+        for (Pair<Character, Character> rang: hebrewRange) {
+            if (rang.first <= c && c <= rang.second) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if the char is in the rtl range, otherwise false
+     */
+    static Boolean isArabic(char c){
+        for (Pair<Character, Character> rang: arabicRange) {
```

Comment:
Similary, I wanted to propose directly using `char` instead of `Character` to avoid all the boxing and unboxing all the time, but from shortly glancing at the code, I cannot tell which one would be better. Can you have a look?

Reply:
Pair can't get char.
Also, there are places I wanted the ability to send null, and that also require Character.

Reply:
Yes, you would simply use a `char[2]` instead of `Pair<Character,Character>`. If you make use of `null` values, you have to be careful, since unboxing a `null` `Character` will lead to a `NullPointerException`.

In any case, please try to use *either* `char` *or* `Character` as far as possible, i.e. avoid autoboxing back and force for performance reasons.

Reply:
I will review it today to see all the cases.
If I don't really need null, then I will change it to char. Otherwise, Character.

Reply:
I need Character for maps, so I switch every char to Character